service/api_logic/user_logic.py

```diff
@@ -128,4 +135,64 @@ async def create_access_token_response(self, user):
         response_json = jsonify(response)
         set_access_cookies(response_json, access_token)
 
-        return response_json
\ No newline at end of file
+        return response_json
+
+
+    def add_preferences(self, dto: UpdateUserPreferencesDTO):
+        if not dto.user_id or not isinstance(dto.preferences, list):
+            raise NotUserIdOrPreferencesError()
+
+        new_dto_by_type_of_preference = self.dto_for_type_of_preference(dto)
+
+        existing_sports = self._preferences_dal.get_all_sport_preference_indexes()
```

Comment:
We should cache this response if not already. I believe we don't expect the response to change often, right?

Maybe SQLAlchemy supports caching it responses. If not, we should probably store this response somewhere ourselves